### PR TITLE
Replacing Mockito statements with own fake socket object

### PR DESCRIPTION
**Pull request by prondzyn**
#417 I replaced Mockito statements in [BkReuseTest.java](https://github.com/prondzyn/takes/blob/417/src/test/java/org/takes/http/BkReuseTest.java) with [MkSocket](https://github.com/prondzyn/takes/blob/417/src/main/java/org/takes/mock/MkSocket.java) according to [Built-in Fake Objects](http://www.yegor256.com/2014/09/23/built-in-fake-objects.html).

### CODE REVIEW

**Comment by davvd** [comment]
@prondzyn Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@dmzaytsev review this one, please

File: src/main/java/org/takes/mock/MkSocket.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+/**
+ * Mock/fake socket.
+ *
+ * @author Piotr Pradzynski (prondzyn@gmail.com)
+ * @version $Id$
+ */
```
Comment (dmzaytsev) [inline src/main/java/org/takes/mock/MkSocket.java]: @prondzyn `since 0.26` I guess

File: src/main/java/org/takes/mock/MkSocketImpl.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.net.SocketImpl;
+
+/**
+ * Mock/fake socket impl.
+ *
+ * @author Piotr Pradzynski (prondzyn@gmail.com)
+ * @version $Id$
+ */
```
Comment (dmzaytsev) [inline src/main/java/org/takes/mock/MkSocketImpl.java]: @prondzyn same here `since 0.26`

File: src/main/java/org/takes/mock/MkSocketImpl.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+class MkSocketImpl extends SocketImpl {
```
Comment (dmzaytsev) [inline src/main/java/org/takes/mock/MkSocketImpl.java]: @prondzyn can we make final class ?
we wouldn't extend this one in the future, isn't it ?

File: src/main/java/org/takes/mock/package-info.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * Mock/fake classes.
+ *
+ * @author Piotr Pradzynski (pradzyn@gmail.com)
+ * @version $Id$
+ */
```
Comment (dmzaytsev) [inline src/main/java/org/takes/mock/package-info.java]: @prondzyn `since 0.26`

File: src/main/java/org/takes/mock/MkSocket.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+public class MkSocket extends Socket {
```
Comment (dmzaytsev) [inline src/main/java/org/takes/mock/MkSocket.java]: @prondzyn same here,  let's make `MkSocket` final

**Comment by dmzaytsev** [comment]
@prondzyn looks good, just 5 minor comments

**Comment by prondzyn** [comment]
@dmzaytsev code improved

**Comment by dmzaytsev** [comment]
@prondzyn thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@prondzyn oops, this looks rather bad. Why creating this mocking version of `Socket` when we can just start a real server using `ServerSocket`? Or using this approach: http://www.yegor256.com/2014/04/18/jcabi-http-server-mocking.html

**Comment by prondzyn** [comment]
@yegor256 please check it now

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, help us to evaluate quality here, as per [par.24](http://www.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@dmzaytsev 
According to our [quality rules](http://www.teamed.io/qa.html):
 > Comments were mostly about design problems, not cosmetic issues.
 
 There were mostly cosmetic issues found during review. Try to pay attention to bigger design problems and the overall quality of the code.
Please confirm that this rule is clear.

**Comment by dmzaytsev** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@dmzaytsev I added 10 mins to @elenavolokhova (for QA review) in transaction 69590230; thanks, paid, **24 mins** to your account, payment ID is `69590235`, task took 128 hours and 17 mins to complete; review comments (c=9) added as a bonus; +24 added to your rating, current score is: [+1335](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**